test/test_torch.py

```diff
@@ -4151,12 +4151,9 @@ def run_test(shape, cast, diagonal):
                             self.assertEqual(x_nc.tril_(diagonal), exp_nc, 0)
 
                         # any 3-dimensional tensor should be fine
-                        if len(shape) <= 3 or s == -2:
+                        if len(shape) <= 3 or s == -2 or s < -3:
```

Comment:
this condition looks bogus, isn't an `s` just an integer?  So `s == -2 or s < -3` is just `s <= -2`.

Also, why do this at all?  Just save whether x_nc is contiguous before the call and check it's the same after.